src/algorithm.rs

```diff
@@ -26,24 +38,115 @@ pub struct WalkResult {
 pub fn random_walk(
     seed_set: Option<SeedSet>,
     network: &NetworkView,
+    damping_factors: DampingFactors,
 ) -> Result<WalkResult, OsrankError> {
-    unimplemented!()
+    match seed_set {
+        Some(_) => unimplemented!(),
+        None => {
+            let mut walks = RandomWalks::new();
+            for i in network.from_graph.node_indices() {
+                // TODO(mb) number of iterations must be a variable
+                for _ in 0..1000 {
+                    let mut walk = RandomWalk::new();
+                    walk.add_next(i);
+                    let mut current_node = i;
+                    // TODO distinguish account/project
+                    // TODO Should there be a safeguard so this doesn't run forever?
+                    while rand::thread_rng().gen::<f64>() < damping_factors.project {
```

Comment:
Yes, I think in the end they are equivalent. I would like to replace it with something more readable in the future anyway, fe something with `choose_weighted` and then it would be.